joomla/joomla-cms
# [4.0] Fix initial value of the Cassiopeia template style parameter "brand" for new installation and update and use it in error.php, too

Pull Request for https://github.com/joomla/joomla-cms/pull/33751#issuecomment-841432541 .

### Summary of Changes

1. Fix the outdated `params` values for the Cassiopeia template style in the base.sql files for new installation so they match to the default values in the XML file, including the changes from #33751 , #31520 and possibly others.

2. Use default value 1 in call to `$this->params->get` for this parameter so it defaults to "Yes" if the values is not set in database e.g. after update from 4.0 Beta 7 or 3.10, which is the same default value as defined in the XML for that parameter.

3. Use the parameter also for the error page as mentioned in the description of PR #33751 :

> The error page can be adapted if this is accepted.

4. Remove the `height: 100vh;` in the `.error_site` block of the `templates/cassiopeia/scss/blocks/_modifiers.scss` to get rid of the otherwise shown blue bar at the top or the error page when no header should be shown, see comment https://github.com/joomla/joomla-cms/pull/33930#issuecomment-842582987 below.

### Testing Instructions

Step 1: Use current 4.0-dev or the nightly build from May 18 or later to make a new installation.

Step 2: After the installation has finished, do not do anything else in backend.

Step 3: In frontend, check if the blue header section is shown.

Result: No blue header section.

Step 4: Apply the changes from this PR.

Step 5: Check again if the blue header section is shown in frontend.

Result: Blue header section with logo "Cassiopeia" and no site description is shown, like it was before PR #33751 was merged, because the default value in the `$this->params->get` call.

Step 6: Remove configuration.php, delete all tables in database and make again a new installation, still with the patch of this PR applied.

Step 7: After the installation has finished, do not do anything else in backend.

Step 8: In frontend, check if the blue header section is shown.

Result: Blue header section with logo "Cassiopeia" and no site description is shown, like it was before PR #33751 was merged, because the parameter has the right value in database.

Step 9: In backend, edit the Cassiopeia template style settings and check if the "Brand" parameter in advanced settings is switched on.

Result: The parameter is switched on.

Step 10: Check the error page by entering an unkown URL in frontend.

Result: Blue header section with logo "Cassiopeia" and no site description is shown.

Step 11: In backend, switch off the "Brand" parameter and save the changes.

Step 12: Check the front page and the error page on frontend if the blue header section is shown.

Result: The blue header is not shown anymore in frontend on normal and error page.

### Actual result BEFORE applying this Pull Request

No blue header section with logo in frontend after a new installation or update from 3.10 or 4.0 Beta 7 and earlier without having saved template style settings.

See https://github.com/joomla/joomla-cms/pull/33751#issuecomment-841432541 .

The error page doesn't respect the "Brand" parameter.

### Expected result AFTER applying this Pull Request

Site shows the header with logo "Cassiopeia" on frontend after a new installation or update without having saved template style settings.

The error page respects the "Brand" parameter.

### Documentation Changes Required

None.
